### PR TITLE
Bump spdlog version and add with_durability (master)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

**Comment by trel** [comment]
there should probably be a new issue for 'create with_durability library', and it link to the original issue https://github.com/irods/irods/issues/3480

**Comment by korydraughn** [comment]
Will create a new issue.

**Comment by trel** [comment]
let us know when you want more eyeballs - there's a lot here

**Comment by korydraughn** [comment]
Tests passed on ubuntu 16.
Not tested on centos.

File: lib/core/include/with_durability.hpp [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+#include <stdexcept>
+#include <thread>
+#include <type_traits>
+#include <utility>
+
+namespace irods::experimental
+{
+    /// A type that specifies how many times \p with_durability should
+    /// retry on failure of a function-like object.
+    struct retries { int value = 2; };
```
Comment (korydraughn) [inline lib/core/include/with_durability.hpp:14]: Default this to 1.

File: lib/core/include/with_durability.hpp [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+    /// A type that specifies a multiplier of \p delay.
+    struct delay_multiplier { float value = 1.f; };
+
+    /// A type that holds options used to control the behavior of \p with_durability.
+    class durability_options
+    {
+    public:
+        constexpr durability_options() noexcept
+            : retries_{2}
```
Comment (korydraughn) [inline lib/core/include/with_durability.hpp:28]: Also default to 1

File: server/api/src/rsDataObjUnlink.cpp [hunk 3]
```diff
@@ -397,15 +358,44 @@ resolveDataObjReplStatus( rsComm_t *rsComm, dataObjInp_t *dataObjUnlinkInp ) {
...
+    {
+        std::string vault_path;
+        if (const auto err = irods::get_vault_path_for_hier_string(dataObjInfo->rescHier, vault_path); !err.ok()) {
+            return SYS_INTERNAL_ERR;
+        }
+
+        if (!has_prefix(dataObjInfo->filePath, vault_path.data())) {
+            dataObjUnlinkInp->oprType = UNREG_OPR;
+
+            logger::api::info("*** REPLICA IS NOT IN A VAULT!!! UNREGISTERING REPLICA INSTEAD *** "
```
Comment (korydraughn) [inline server/api/src/rsDataObjUnlink.cpp:385]: Tell them that we are leaving it in-place on disk.

**Comment by korydraughn** [comment]
Updated and added #

**Pull request merged**